test/scaling_test/scaling_test.go

```diff
@@ -0,0 +1,398 @@
+
+			// Scale Standalone instance
+			testenvInstance.Log.Info("Scaling Up Standalone CR")
+			scaledReplicaCount := 2
+			standalone = &enterprisev1.Standalone{}
+			err = deployment.GetInstance(deployment.GetName(), standalone)
+			Expect(err).To(Succeed(), "Failed to get instance of Standalone")
+
+			standalone.Spec.Replicas = int32(scaledReplicaCount)
+
+			err = deployment.UpdateCR(standalone)
+			Expect(err).To(Succeed(), "Failed to scale up Standalone")
+
+			// Ensure standalone is scaling up
+			testenv.VerifyStandalonePhase(deployment, testenvInstance, deployment.GetName(), splcommon.PhaseScalingUp)
+
+			// Wait for Standalone to be in READY status
+			testenv.VerifyStandalonePhase(deployment, testenvInstance, deployment.GetName(), splcommon.PhaseReady)
+
+			// Wait for Monitoring Console Pod to be in READY status
+			testenv.MCPodReady(testenvInstance.GetName(), deployment)
+
+			// Scale Down Standalone
+			testenvInstance.Log.Info("Scaling Down Standalone CR")
+			scaledReplicaCount = scaledReplicaCount - 1
+			standalone = &enterprisev1.Standalone{}
+			err = deployment.GetInstance(deployment.GetName(), standalone)
+			Expect(err).To(Succeed(), "Failed to get instance of Standalone")
+
+			standalone.Spec.Replicas = int32(scaledReplicaCount)
+
+			err = deployment.UpdateCR(standalone)
+			Expect(err).To(Succeed(), "Failed to scale down Standalone")
+
+			// Ensure standalone is scaling down
+			testenv.VerifyStandalonePhase(deployment, testenvInstance, deployment.GetName(), splcommon.PhaseScalingDown)
+
+			// Wait for Standalone to be in READY status
+			testenv.StandaloneReady(deployment, deployment.GetName(), standalone, testenvInstance)
+
+			// Wait for Monitoring Console Pod to be in READY status
+			testenv.MCPodReady(testenvInstance.GetName(), deployment)
+		})
+	})
+
+	Context("Clustered deployment (C3 - clustered indexer, search head cluster)", func() {
+		It("scaling_test: SHC and IDXC can be scaled up and data is searchable", func() {
+
+			defaultSHReplicas := 3
+			defaultIndexerReplicas := 3
+			err := deployment.DeploySingleSiteCluster(deployment.GetName(), defaultIndexerReplicas, true)
+			Expect(err).To(Succeed(), "Unable to deploy search head cluster")
+
+			// Ensure that the cluster-master goes to Ready phase
+			testenv.ClusterMasterReady(deployment, testenvInstance)
+
+			// Ensure indexers go to Ready phase
+			testenv.SingleSiteIndexersReady(deployment, testenvInstance)
+
+			// Ensure search head cluster go to Ready phase
+			testenv.SearchHeadClusterReady(deployment, testenvInstance)
+
+			// Wait for Monitoring Console Pod to be in READY status
+			testenv.MCPodReady(testenvInstance.GetName(), deployment)
+
+			// Scale Search Head Cluster
+			scaledSHReplicas := defaultSHReplicas + 1
+			testenvInstance.Log.Info("Scaling up Search Head Cluster", "Current Replicas", defaultSHReplicas, "New Replicas", scaledSHReplicas)
+			shcName := deployment.GetName() + "-shc"
```

Comment:
Should we maintain the extensions "-shc" and "-idc" as constant strings? Or have a formatted string such as:

const string indexerNameBuilder = %s-idc
const string shcNameBuilder = %s-shc

Reply:
No @akondur this controlled by a method in deployment.go so the naming convention is not strictly imposed. I will make another PR with standardization and refactor of test code and take care of these issues then.